src/sandstorm/web-session.capnp

```diff
@@ -97,6 +107,32 @@ interface WebSession @0xa50711a14d35a8ce extends(Grain.UiSession) {
 
     accept @2 :List(AcceptedType);
     # This corresponds to the Accept header
+
+    additionalHeaders @3 :List(Header);
+    # Additional headers present in the request. Only whitelisted headers are
+    # permitted.
+
+    struct Header {
+      name @0 :Text;  # lower-cased name
+      value @1 :Text;
+    }
+
+    const headerWhitelist :List(Text) = [
+      "if-match",              # webdav etag
+      "if-not-match",          # webdav etag
+      "depth",                 # webdav PROPFIND
+      "destination",           # webdav COPY/MOVE
+      "overwrite",             # webdav COPY/MOVE
+      "oc-total-length",       # Owncloud client
+      "oc-chunk-size",         # Owncloud client
+      "x-oc-mtime",            # Owncloud client
+      "oc-fileid",             # Owncloud client
+      "oc-chunked",            # Owncloud client
+    ];
+    # Non-standard request headers which are whitelisted for backwards-compatibility
```

Comment:
This comment should go on the line after `headerWhitelist` is declared, i.e. inside the [], like Python doc comments:

    const headerWhitelist :List(Text) = [
      # Non-standard request headers which are whitelisted for backwards-compatibility
      # ...

      "if-match",              # webdav etag
      ...
    ];